protocol-units/bridge/chains/movement/src/lib.rs

```diff
@@ -0,0 +1,198 @@
+	///The signer account
+	signer: Arc<LocalAccount>,
+}
+
+impl MovementClient {
+	pub async fn new(config: Config) -> Result<Self, anyhow::Error> {
+		let dot_movement = dot_movement::DotMovement::try_from_env().unwrap();
+		let suzuka_config =
+			dot_movement.try_get_config_from_json::<suzuka_config::Config>().unwrap();
+		let node_connection_address = suzuka_config
+			.execution_config
+			.maptos_config
+			.client
+			.maptos_rest_connection_hostname;
+		let node_connection_port =
+			suzuka_config.execution_config.maptos_config.client.maptos_rest_connection_port;
+
+		let node_connection_url =
+			format!("http://{}:{}", node_connection_address, node_connection_port);
+		let node_connection_url = Url::from_str(node_connection_url.as_str()).unwrap();
+
+		let rest_client = Client::new(node_connection_url.clone());
+
+		let seed = [3u8; 32];
+		let mut rng = rand::rngs::StdRng::from_seed(seed);
+		let signer = LocalAccount::generate(&mut rng);
+
+		Ok(MovementClient {
+			initiator_address: Vec::new(), //dummy for now
+			rest_client,
+			counterparty_address: DUMMY_ADDRESS,
+			signer: Arc::new(signer),
+		})
+	}
+}
+
+#[async_trait::async_trait]
+impl BridgeContractCounterparty for MovementClient {
+	type Address = MovementAddress;
+	type Hash = [u8; 32];
+
+	async fn lock_bridge_transfer_assets(
+		&mut self,
+		bridge_transfer_id: BridgeTransferId<Self::Hash>,
+		hash_lock: HashLock<Self::Hash>,
+		time_lock: TimeLock,
+		initiator: InitiatorAddress<Vec<u8>>,
+		recipient: RecipientAddress<Self::Address>,
+		amount: Amount,
+	) -> BridgeContractCounterpartyResult<()> {
+		//@TODO properly return an error instead of unwrapping
+		let args = vec![
+			to_bcs_bytes(&initiator.0).unwrap(),
+			to_bcs_bytes(&bridge_transfer_id.0).unwrap(),
+			to_bcs_bytes(&hash_lock.0).unwrap(),
+			to_bcs_bytes(&time_lock.0).unwrap(),
+			to_bcs_bytes(&recipient.0).unwrap(),
+			to_bcs_bytes(&amount.0).unwrap(),
+		];
+		let payload = utils::make_aptos_payload(
+			self.counterparty_address,
+			COUNTERPARTY_MODULE_NAME,
+			"lock_bridge_transfer_assets",
+			self.counterparty_type_args(Call::Lock),
+			args,
+		);
+		let _ = utils::send_aptos_transaction(&self.rest_client, self.signer.as_ref(), payload)
+			.await
+			.map_err(|_| BridgeContractCounterpartyError::LockTransferAssetsError);
+		Ok(())
+	}
+
+	async fn complete_bridge_transfer(
+		&mut self,
+		bridge_transfer_id: BridgeTransferId<Self::Hash>,
+		preimage: HashLockPreImage,
+	) -> BridgeContractCounterpartyResult<()> {
+		let args = vec![
+			to_bcs_bytes(&self.signer.address()).unwrap(),
+			to_bcs_bytes(&bridge_transfer_id.0).unwrap(),
+			to_bcs_bytes(&preimage.0).unwrap(),
+		];
+		let payload = utils::make_aptos_payload(
+			self.counterparty_address,
+			COUNTERPARTY_MODULE_NAME,
+			"complete_bridge_transfer",
+			self.counterparty_type_args(Call::Complete),
+			args,
+		);
+
+		let _ = utils::send_aptos_transaction(&self.rest_client, self.signer.as_ref(), payload)
```

Comment:
Is this actually sending and awaiting? One way or another the names are not lining up here. If it's sending the Aptos transaction only, then I wouldn't call this method `complete_bridge_transfer` as we don't have notification of completion. If it's sending and awaiting confirmation, then it should say that.

Reply:
Yes it's submitting and waiting. Name has been updated to improve semantics here.